core/Network/TLS/Parameters.hs

```diff
@@ -141,6 +153,58 @@ defaultParamsClient serverName serverId = ClientParams
     , clientEarlyData               = Nothing
     }
 
+-- | Loads the client cert to provide when the server requests client authentication.
+clientParamsSetClientCert :: FilePath -> FilePath -> ClientParams -> IO (Either String ClientParams)
+clientParamsSetClientCert keyFile certificateFile params = do
+    keys <- E.handle (\(_ :: E.IOException) -> pure []) $ readKeyFile keyFile
```

Comment:
This benefits from https://github.com/vincenthz/hs-certificate/pull/116 (otherwise that failure won't be caught here)